ToolBox/KaikoDataConverter/KaikoDataConverterProgram.cs

```diff
@@ -13,42 +13,103 @@
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using ZipFile = Ionic.Zip.ZipFile;
 
 namespace QuantConnect.ToolBox.KaikoDataConverter
 {
     /// <summary>
-    /// Console application for converting Kaiko data into Lean data format
+    /// Console application for converting a single day of Kaiko data into Lean data format for high resolutions (tick, second and minute)
     /// </summary>
     public static class KaikoDataConverterProgram
     {
-        public static void KaikoDataConverter(string market, string _tickType, string sourceDirectory)
+        /// <summary>
+        /// Kaiko data converter entry point.
+        /// </summary>
+        /// <param name="sourceDirectory">The source directory where all Kaiko zipped files are stored..</param>
+        /// <param name="date">The date to process.</param>
+        /// <exception cref="ArgumentException">Source folder does not exists.</exception>
+        /// <remarks>This converter will process automatically data for every exchange and for both tick types if the raw data files are available in the sourceDirectory</remarks>
+        public static void KaikoDataConverter(string sourceDirectory, string date)
         {
-            if (market.IsNullOrEmpty() || _tickType.IsNullOrEmpty() || sourceDirectory.IsNullOrEmpty())
+            var timer = new Stopwatch();
+            timer.Start();
+            var folderPath = new DirectoryInfo(sourceDirectory);
+            if (!folderPath.Exists)
             {
-                throw new ArgumentException("The arguments must be specified as [market] [tickType (quote/trade)] [kaiko raw data directory path]");
+                throw new ArgumentException($"Source folder {folderPath.FullName} not found");
             }
-            var tickType = _tickType.ToLower() == "quote" ? TickType.Quote : TickType.Trade;
 
-            if (tickType == TickType.Quote)
-            {
-                CreateCryptoTicks(sourceDirectory, market, tickType, ParseKaikoQuoteFile);
-                AggregateTicksInAllResolutions(market, tickType);
-            }
-            else
+            foreach (var filePath in folderPath.EnumerateFiles("*.zip"))
             {
-                CreateCryptoTicks(sourceDirectory, market, tickType, ParseKaikoTradeFile);
-                AggregateTicksInAllResolutions(market, tickType);
+                var processingDate = DateTime.ParseExact(date, DateFormat.EightCharacter, CultureInfo.InvariantCulture);
+                Log.Trace($"KaikoDataConverter(): Starting data conversion from source {filePath.Name} for date {processingDate:yyyy_MM_dd}... ");
+                var zip = new ZipFile(filePath.FullName);
+                var targetDayEntries = zip.Entries.Where(e => e.FileName.Contains($"{processingDate:yyyy_MM_dd}"));
+
+                if (targetDayEntries.Count() == 0)
+                {
+                    Log.Error($"KaikoDataConverter(): Date {processingDate:yyyy_MM_dd} not found in source file.");
```

Comment:
maybe you could log the file name, since its looping through all files in the given directory